packages/openapi-v3/src/__tests__/unit/enhancers/consolidate.spec.extension.unit.ts

```diff
@@ -0,0 +1,166 @@
+              },
+            },
+          },
+        }),
+      )
+      .withComponents(
+        new ComponentsSpecBuilder().withSchema('loopback.example', {
+          title: 'loopback.example',
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        }),
+      )
+      .build();
+
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(expectedSpec);
+  });
+
+  it('ignores schema without title property', () => {
+    const inputSpec = new OpenApiSpecBuilder()
+      .withOperation(
+        'get',
+        '/',
+        new OperationSpecBuilder().withResponse(200, {
+          description: 'Example',
+          content: {
+            'application/json': {
+              schema: {
+                properties: {
+                  test: {
+                    type: 'string',
+                  },
+                },
+              },
+            },
+          },
+        }),
+      )
+      .build();
+
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(inputSpec);
+  });
+
+  it('Avoids naming collision', () => {
+    const inputSpec = new OpenApiSpecBuilder()
+      .withOperation(
+        'get',
+        '/',
+        new OperationSpecBuilder().withResponse(200, {
+          description: 'Example',
+          content: {
+            'application/json': {
+              schema: {
+                title: 'loopback.example',
+                properties: {
+                  test: {
+                    type: 'string',
+                  },
+                },
+              },
+            },
+          },
+        }),
+      )
+      .withComponents(
+        new ComponentsSpecBuilder().withSchema('loopback.example', {
+          title: 'Different loopback.example exists',
+          properties: {
+            testDiff: {
+              type: 'string',
+            },
+          },
+        }),
+      )
+      .build();
+
+    const expectedSpec = new OpenApiSpecBuilder()
+      .withOperation(
+        'get',
+        '/',
+        new OperationSpecBuilder().withResponse(200, {
+          description: 'Example',
+          content: {
+            'application/json': {
+              schema: {
+                $ref: '#/components/schemas/loopback.example1',
+              },
+            },
+          },
+        }),
+      )
+      .withComponents(
+        new ComponentsSpecBuilder()
+          .withSchema('loopback.example', {
+            title: 'Different loopback.example exists',
+            properties: {
+              testDiff: {
+                type: 'string',
+              },
+            },
+          })
+          .withSchema('loopback.example1', {
+            title: 'loopback.example',
+            properties: {
+              test: {
+                type: 'string',
+              },
+            },
+          }),
+      )
+      .build();
+
+    expect(consolidationEnhancer.modifySpec(inputSpec)).to.eql(expectedSpec);
+  });
```

Comment:
(Feel free to leave it out of the scope of this PR, I know it's exhausting to be blocked by such minor changes :p and appreciate your patience again!)

Maybe add one test for consolidating the same schema that exists multiple times into one `component/schema/<schema_name>`?

```ts
// The input spec
const inputSpec = new OpenApiSpecBuilder()
      .withOperation(
        'get',
        // first time has 'loopback.example' 
        '/path1',
        new OperationSpecBuilder().withResponse(200, {
          description: 'Example',
          content: {
            'application/json': {
              schema: {
                title: 'loopback.example',
                properties: {
                  test: {
                    type: 'string',
                  },
                },
              },
            },
          },
        }),
      )
.withOperation(
        'get',
        // second time has 'loopback.example' 
        '/path2',
        new OperationSpecBuilder().withResponse(200, {
          description: 'Example',
          content: {
            'application/json': {
              schema: {
                title: 'loopback.example',
                properties: {
                  test: {
                    type: 'string',
                  },
                },
              },
            },
          },
        }),
      )
.build()
```

and

```ts
const expectedSpec = new OpenApiSpecBuilder()
      .withOperation(
        'get',
        '/path1',
        new OperationSpecBuilder().withResponse(200, {
          description: 'Example',
          content: {
            'application/json': {
              schema: {
                $ref: '#/components/schemas/loopback.example',
              },
            },
          },
        }),
      )
   .withOperation(
        'get',
        '/path2',
        new OperationSpecBuilder().withResponse(200, {
          description: 'Example',
          content: {
            'application/json': {
              schema: {
                $ref: '#/components/schemas/loopback.example',
              },
            },
          },
        }),
      )
      .withComponents(
        new ComponentsSpecBuilder()
          .withSchema('loopback.example', {
            title: 'loopback.example',
            properties: {
              test: {
                type: 'string',
              },
            },
          }),
      )
      .build();
```

Reply:
Added.  Thank you!